clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2867,7 +2958,10 @@ namespace {
       BoundsExpr *B = nullptr;
       InteropTypeExpr *IT = nullptr;
       if (VD) {
-        B = VD->getBoundsExpr();
+        if (State.UC[VD])
```

Comment:
Should use the find method instead of the subscript operator?   This avoids two lookups in the DenseMap.